### PR TITLE
Stroke palette

### PR DESCRIPTION
**Pull request by karlh18**
- Refactored Code
- Added two Unit Tests

### CODE REVIEW

**Review by prad1998 — COMMENTED** [review]
Delete comment

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java [hunk 1]
```diff
@@ -63,255 +66,233 @@ public void setEditor(DrawingEditor newValue) {
...
-                    gbc.gridy = 1;
-                    gbc.insets = new Insets(3, 3, 0, 0);
-                    p.add(btn, gbc);
-                    
-                    // Create dash offset field and dashes button
-                    JAttributeTextField<Double> dashOffsetField = new JAttributeTextField<Double>();
-                    dashOffsetField.setColumns(1);
-                    dashOffsetField.setToolTipText(labels.getString("attribute.strokeDashPhase.toolTipText"));
-                    dashOffsetField.setHorizontalAlignment(JAttributeTextField.LEFT);
-                    //dashOffsetField.putClientProperty("Palette.Component.segmentPosition", "first");
```
Comment (prad1998) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java:290]: Delete comment

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java [hunk 2]
```diff
@@ -63,255 +66,233 @@ public void setEditor(DrawingEditor newValue) {
...
+     *
+     * @param state Depending on the state if should add with field or not
+     * @return return the JPanel with the added buttons
+     */
     @Override
     @FeatureEntryPoint(JHotDrawFeatures.STROKE_PALETTE)
     protected JComponent createDisclosedComponent(int state) {
-        JPanel p = null;
-
+        ResourceBundleUtil labels = ResourceBundleUtil.getBundle("org.jhotdraw.samples.svg.Labels");
```
Comment (prad1998) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java:79]: Maybe not necessary to initialize it again as it is already done in the constructor

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java [hunk 3]
```diff
@@ -63,255 +66,233 @@ public void setEditor(DrawingEditor newValue) {
...
-                    gbc.insets = new Insets(3, 3, 0, 0);
-                    gbc.fill = GridBagConstraints.BOTH;
-                    gbc.gridwidth=2;
-                    p.add(dashOffsetField, gbc);
+    void createStrokeDashField(ResourceBundleUtil labels) {
+        JAttributeTextField<Double> dashOffsetField = new JAttributeTextField<Double>();
+        dashOffsetField.setColumns(1);
+        dashOffsetField.setToolTipText(labels.getString("attribute.strokeDashPhase.toolTipText"));
+        dashOffsetField.setHorizontalAlignment(JAttributeTextField.LEFT);
+        //dashOffsetField.putClientProperty("Palette.Component.segmentPosition", "first");
```
Comment (prad1998) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java:271]: Delete comment

File: JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/StrokeToolBarTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    public static void setUpClass() {
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+    }
+
+    @Before
+    public void setUp() {
+        //TODO : Change to mocks
```
Comment (prad1998) [inline JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/StrokeToolBarTest.java:42]: Delete TODO

File: JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/StrokeToolBarTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        strokeToolBar = null;
+    }
+
+    /**
+     * Test of createDisclosedComponent method, of class StrokeToolBar.
+     */
+    @Test
+    public void testCreateDisclosedComponent() {
+
+        //TODO : Change to mocks
```
Comment (prad1998) [inline JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/StrokeToolBarTest.java:62]: Delete TODO

File: JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/WhenAlterFigures.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    @ProvidedScenarioState
+    DrawingEditor editor;
+
+    @ProvidedScenarioState
+    Set<Figure> selectedFigures;
+
+    @BeforeStage
+    private void before() {
+        selectedFigures = new HashSet<>(editor.getActiveView().getSelectedFigures());
+        //selectedFigures.iterator().next().getAttribute(STROKE_JOIN);
```
Comment (prad1998) [inline JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/WhenAlterFigures.java:37]: Delete comment

File: JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/GivenFigureToAlterLines.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+public class GivenFigureToAlterLines extends Stage<GivenFigureToAlterLines> {
+
+    @ProvidedScenarioState
+    DrawingEditor editor;
+
+    @ProvidedScenarioState
+    Figure figure;
+
+    @ProvidedScenarioState
+    Map<AttributeKey, Object> attribute;
```
Comment (prad1998) [inline JHotDraw/src/test/java/org/jhotdraw/samples/svg/gui/GivenFigureToAlterLines.java:26]: Is this attribute used?

**Review by prad1998 — COMMENTED** [review]
Overall fine. There are some unused attributes that could have been deleted and also a few comments.

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java (hunk 2 above)
Comment (karlh18) [reply JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/StrokeToolBar.java:79]: It is only on the constructor scope, which is why it is  created anew in the method.

**Review by prad1998 — APPROVED** [review]

**Pull request merged**